torch/csrc/jit/passes/quantization.cpp

```diff
@@ -18,8 +18,30 @@ void PropagateQuantInfo(std::shared_ptr<Graph>& graph) {
   throw std::runtime_error("Pass not implemented yet!");
 }
 
+static void addObserverFor(torch::jit::Value* v) {
+  torch::jit::Node* def = v->node();
+  torch::jit::Node* observer =
+      def->owningGraph()->create(at::Symbol::fromQualString("aten::observer"));
+  def->owningBlock()->appendNode(observer);
+  observer->moveAfter(def);
+  v->replaceAllUsesWith(observer->outputs()[0]);
+  observer->outputs()[0]->setUniqueName(v->uniqueName() + ".observed");
+  observer->addInput(v);
+}
+
 void InsertObserverNodes(std::shared_ptr<Graph>& graph) {
-  throw std::runtime_error("Pass not implemented yet!");
+  for (const auto& n : graph->nodes()) {
+    // Skip nodes that we've just added
+    if (n->kind().toQualString() == std::string("aten::observer")) {
+      continue;
+    }
+
+    // Add observers to all outputs of node N unless they are inside a subgraph
+    // that we expect to be fused.
+    for (torch::jit::Value* v : n->outputs()) {
+      addObserverFor(v);
+    }
+  }
```

Comment:
Are you intending to skip all subblocks?

Reply:
We will need to visit subblocks as well, good catch. I'll update the PR accordingly (I'll also need to refine its logic to e.g. work correctly with `prim::` nodes).